sample-compose/app/src/androidTest/java/co/nimblehq/sample/compose/ui/screens/home/HomeScreenTest.kt

```diff
@@ -0,0 +1,45 @@
+package co.nimblehq.sample.compose.ui.screens.home
+
+import androidx.activity.compose.setContent
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.GrantPermissionRule
+import co.nimblehq.sample.compose.ui.AppDestination
+import co.nimblehq.sample.compose.ui.screens.MainActivity
+import org.junit.*
+import org.junit.runner.RunWith
+
+@ExperimentalTestApi
+@RunWith(AndroidJUnit4::class)
+class HomeScreenTest {
+
+    @get:Rule
+    val composeRule = createAndroidComposeRule<MainActivity>()
+
+    /**
+     * More test samples with Runtime Permissions https://alexzh.com/ui-testing-of-android-runtime-permissions/
+     */
+    @get:Rule
+    val permissionRule: GrantPermissionRule = GrantPermissionRule.grant(
+        android.Manifest.permission.CAMERA
+    )
+
+    private var expectedAppDestination: AppDestination? = null
+
+    @Before
+    fun setup() {
+        composeRule.activity.setContent {
+            HomeScreen(
+                navigator = { destination -> expectedAppDestination = destination }
+            )
+        }
+    }
+
+    @Test
+    fun when_entering_the_Home_screen__it_shows_UI_correctly() {
```

Comment:
I am not so sure about this testing function naming convention @luongvo 🤔 May I ask why don't we use it normally as below:
```suggestion
    fun `When entering the Home screen, it shows UI correctly`() {
```

Reply:
@thiennguyen0196 I think we should NOT use backtick in instrument tests (androidTest).

- From the official documentation https://developer.android.com/training/testing/instrumented-tests#create-instrumented
> Using backticks to name tests in Kotlin is only supported on devices running API 30 and above.

<img width="879" alt="image" src="https://user-images.githubusercontent.com/16315358/217431574-e59306d0-dbcd-4c26-b669-032938a381b6.png">

So running the instrument tests on devices with API >= 30 will pass, but we can not run them on devices with API < 30:

More info https://stackoverflow.com/a/62597742
![image](https://user-images.githubusercontent.com/16315358/217433969-c39cc78e-3561-4c6e-a6f3-8b77ac24ee98.png)

<img width="676" alt="image" src="https://user-images.githubusercontent.com/16315358/217434671-bc3e21b5-c848-4b4b-b2f1-8da35ae7cb31.png">


- From the Kotlin convention, it says we can only use backtick in tests, not Android runtime https://kotlinlang.org/docs/coding-conventions.html#names-for-test-methods.

Reply:
@luongvo I see, thanks for clarifying it to me 👏